quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -70,18 +136,16 @@ export class CompilerVisitor implements IRVisitor {
   //  - an instance of Register
   //  - an instance of Callable.
   // The keys should be constructed via `kindName`.
-  private context: Map<string, Computable> = builtinContext()
+  private context: Map<string, Computable>
 
   // all variables declared during compilation
-  private vars: Register[] = []
+  private vars: Register[]
   // the registers allocated for the next-state values of vars
-  private nextVars: Register[] = []
+  private nextVars: Register[]
   // shadow variables that are used by the simulator
-  private shadowVars: Register[] = []
-  // messages that are produced during compilation
-  private compileErrors: ir.IrErrorMessage[] = []
-  // messages that get populated as the compiled code is executed
-  private runtimeErrors: ir.IrErrorMessage[] = []
+  private shadowVars: Register[]
+  // keeps errors in a state
+  private errorTracker: CompilerErrorTracker
```

Comment:
I think that would make things worse. Every time we want to access `context`, for example, we'd have to type/read `this.evaluationState.context` instead of just `this.context`. Also, we'd never use the entire `this.evaluationState` in the internal methods - that combination of attributes is only used in external facing interfaces. So I think it makes much more sense to just spread out this interface like I'm doing now for ease of access internally, and only wrap it inside an `EvaluationState` in the external facing interfaces.

Reply:
sounds right, thanks for considering :)